R/write_eml.R

```diff
@@ -96,13 +96,10 @@ write_eml <- function(x, directory, derived_paragraph = TRUE) {
     maintenanceUpdateFrequency = "unknown"
   )
 
-  # Convert contributors to a data frame
+  # Create creators
   orcid_regex <- "(\\d{4}-){3}\\d{3}(\\d|X)"
   creators <-
-    purrr::map_dfr(
-      x$contributors,
-      ~ as.data.frame(., stringsAsFactors = FALSE)
-    ) %>%
+    contributors(x) %>%
     dplyr::filter(!.data$role %in% c("rightsHolder", "publisher")) %>%
```

Comment:
We'll have to make sure this works if no `role` is assigned. Guaranteeing columns should be done in `contributors()`